src/rdds/lib/processpool/process_pool.py

```diff
@@ -0,0 +1,227 @@
+import logging
+import threading
+from queue import Queue as Queue
+from threading import Thread, Lock, Event
+from multiprocessing import Process, ProcessError, cpu_count, log_to_stderr, get_context
+from multiprocessing.context import SpawnContext
+from typing import *
+from time import sleep
+from dataclasses import dataclass
+
+LOGGER = log_to_stderr()
+LOGGER.setLevel(logging.INFO)
+
+
+@dataclass
+class DummyProcess:
+    name: str
+    exitcode: int
+
+
+@dataclass
+class Task:
+    args: Iterable
+    process: Union[Process, DummyProcess]
+
+
+CompletedTaskQueue = Queue
+
+
+class ProcessPool:
+
+    """
+    Implements a simple multiprocessing pool supporting synchronous and asynchronous processing mode.
+    """
+
+    def __init__(self,
+                 fn: Callable,
+                 args: Iterable,
+                 process_names: List[str] = None,
+                 workers: int = cpu_count() - 1  # Reserve a core for main process
+                 ):
+        """
+        :param fn: The function to execute concurrently
+        :param args: Function arguments
+        :param process_names: List of names to assign to processes for traceability
+        :param workers: Maximum concurrent workers running 'fn' simultaneously
+        """
+        self._fn = fn
+        self._args = args
+        self._n_expected_tasks: int = len(self._args)
+        self._process_names = process_names
+        if self._process_names is None:
+            self._process_names: List[Any] = [None] * self._n_expected_tasks
```

Comment:
```suggestion
        self._process_names = process_names if process_names else [None] * self._n_expected_tasks
```
Is this [None] a fancy way of writing 1?